mysql-test/main/rpl_mysqldump_slave.test

```diff
@@ -86,3 +86,52 @@ DROP TABLE t2;
 
 
 --source include/rpl_end.inc
+
+#
+# MDEV-32611 Added test for mysqldump --delete-master-logs option
+#
+
+connection master;
+use test;
+
+--source include/have_binlog_format_row.inc
+
+use test;
+
+CREATE TABLE t (
+    id int,
+    name varchar(255),
+    male boolean
+);
+
+insert into t values (1, 'John', true);
+insert into t values (2, 'Mary', false);
+
+SHOW MASTER STATUS;
+
+--let $predump_binlog_filename= query_get_value(SHOW MASTER STATUS, File, 1)
+--let $predump_binlog_pos= query_get_value(SHOW MASTER STATUS, Position, 1)
+
+# Execute mysqldump with delete-master-logs option
+--exec $MYSQL_DUMP --compact --delete-master-logs test
+
+--let $postdump_binlog_filename= query_get_value(SHOW MASTER STATUS, File, 1)
+--let $postdump_binlog_pos= query_get_value(SHOW MASTER STATUS, Position, 1)
+
+SHOW MASTER STATUS;
+
+if ($predump_binlog_filename != $postdump_binlog_filename)
+{
+  --echo # predump_binlog_filename: $predump_binlog_filename
+  --echo # postdump_binlog_filename: $postdump_binlog_filename
+  --echo # Master state changed after mariadb-dump with --delete-master-logs
+}
+
+if ($predump_binlog_pos != $postdump_binlog_pos)
+{
+  --echo # predump_binlog_pos: $predump_binlog_pos
+  --echo # postdump_binlog_pos: $postdump_binlog_pos
+  --echo # Master state changed after mariadb-dump with --delete-master-logs
+}
+
```

Comment:
I think we should add another assertion to ensure that the actual log deletion happened, in addition to the log state change verification. To do so, after the `$MYSQL_DUMP`, we should save the log name of the first row of `SHOW BINARY LOGS` (with something like `--let $postdump_first_binary_log_file= query_get_value(SHOW BINARY LOGS, Log_name, 1)`), which should be the first row present on the master, and compare it with the log name of `SHOW MASTER STATUS`, and ensure they are the same (and `--die` if they are different).

Reply:
compared `--let $postdump_first_binary_log_filename= query_get_value(SHOW BINARY LOGS, Log_name, 1)` with pre-existing `$postdump_binlog_filename`